index.js

```diff
@@ -207,7 +208,7 @@ module.exports = async function (url, options) {
                 const waitTime = getRetryDelay(retryOptions);
                 if (getTimeRemaining(retryOptions) < waitTime) {
                     console.error(`Timeout during retry delay, returning last received response`);
-                    resolve(response);
+                    processLastResult();
```

Comment:
I wonder if this will have side-effects... THis might trigger failures in unit tests/ITs using node-fetch-retry. Let's see.

Reply:
So the idea is we might have a failure or a success in the last response, and so I have to capture that as a lambda function to replay in case there is not enough time for another retry attempt.   I might be able to clean this up a little better.

Reply:
OK this is now no longer using a lambda anymore, I just pushed the calculation of time remaining for wait into the method that answers the question if we should retry.